utils/joiner/joinpartition.h

```diff
@@ -164,7 +164,7 @@ class JoinPartition
 
     /* Compression support */
     bool useCompression;
-    compress::IDBCompressInterface compressor;
+    std::shared_ptr<compress::CompressInterface> compressor;
```

Comment:
`std::unique_ptr` isn't enough here?

Reply:
It has the copy constructor, https://github.com/mariadb-corporation/mariadb-columnstore-engine/blob/develop/utils/joiner/joinpartition.cpp#L109 as far as I know we cannot copy `unique_ptr`, please fix me if I'm wrong.

Reply:
Yes, you're right, I missed that point.